### PR TITLE
Added helpers for logs and pods in deployment

### PR DESCRIPTION
**Pull request by AverageMarcus**
Added a couple new helper functions:
* Get all pods associated with a given Deployment
* Get logs from a given Pod

### CODE REVIEW

File: pkg/client/logs.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// GetLogs fetches the logs from the provided Pod. If `numOfLines` is provided (instead of `nil`) then that
+// many lines will be returned from the end of the logs.
+func (c *Client) GetLogs(ctx context.Context, pod *corev1.Pod, numOfLines *int64) (string, error) {
+	coreClient, err := kubernetes.NewForConfig(c.config)
+	if err != nil {
+		return "", fmt.Errorf("failed initializing kubernetes core client - %v", err)
```
Comment (glitchcrab) [inline pkg/client/logs.go:18]: my inner perfectionist wants Kubernetes to be capitalised, but that's just me being super picky :sweat_smile:

**Review by glitchcrab — APPROVED** [review]
please accept my (very beginner) review: LGTM

File: pkg/client/logs.go (hunk 1 above)
Comment (AverageMarcus) [reply pkg/client/logs.go:18]: I actually copied this same couple lines from another area of the code so don't want to have it inconsistent in this PR. might sort it everywhere later.

Comment (glitchcrab) [reply pkg/client/logs.go:18]: that is an acceptable reason. Wealdy approves this message.

**Pull request merged**